cmd/tfplugindocs/testdata/scripts/generate/framework_provider_success_generic_templates.txtar

```diff
@@ -0,0 +1,928 @@
+	// httpResp, err := r.client.Do(httpReq)
+	// if err != nil {
+	//     resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create example, got error: %s", err))
+	//     return
+	// }
+
+	// For the purposes of this example code, hardcoding a response value to
+	// save into the Terraform state.
+	data.Id = types.StringValue("example-id")
+
+	// Write logs using the tflog package
+	// Documentation: https://terraform.io/plugin/log
+	tflog.Trace(ctx, "created a resource")
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ExampleResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data ExampleResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If applicable, this is a great opportunity to initialize any necessary
+	// provider client data and make a call using it.
+	// httpResp, err := r.client.Do(httpReq)
+	// if err != nil {
+	//     resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read example, got error: %s", err))
+	//     return
+	// }
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ExampleResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data ExampleResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If applicable, this is a great opportunity to initialize any necessary
+	// provider client data and make a call using it.
+	// httpResp, err := r.client.Do(httpReq)
+	// if err != nil {
+	//     resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to update example, got error: %s", err))
+	//     return
+	// }
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ExampleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data ExampleResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If applicable, this is a great opportunity to initialize any necessary
+	// provider client data and make a call using it.
+	// httpResp, err := r.client.Do(httpReq)
+	// if err != nil {
+	//     resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete example, got error: %s", err))
+	//     return
+	// }
+}
+
+func (r *ExampleResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+-- internal/provider/provider.go --
```

Comment:
> How is the content created for new tests?

For the first couple of tests, I had a separate repo of the provider in the wanted test conditions and ran the `txtar` [CLI](https://pkg.go.dev/golang.org/x/exp/cmd/txtar) to create a `.txtar` representation of the repo. I then ran the `tfplugindocs` on the source repo and copy/paste or run `txtar` on the resulting docs to create the golden files. 

> How is existing content updated?

it is possible to "extract" `.txtar` files into back into it's original state, although overall it would be a time consuming process. Short-term since most changes to the test would be altering the doc templates or golden files, it wouldn't be too bad for quick changes, but there would be a lot of overhead for long-term maintenance if we would have to change the provider source files. 

Overall, I agree with your assessment that refactoring the code and separating test concerns would be the best case scenario. If we do that then, we can significantly reduce the length of the test scripts and would have to worry less about overall maintainability.

Reply:
Excellent! It might be helpful to capture some of this information in the contributing guide so everyone can know what they need to do so they can work with the acceptance testing suite, e.g. it sounds like we should be running `go install golang.org/x/exp/cmd/txtar@latest` so we have that CLI available locally. 👍